libafl_qemu/src/emu.rs

```diff
@@ -673,28 +673,44 @@ pub struct Emulator {
     _private: (),
 }
 
+#[derive(Debug, thiserror::Error)]
+pub enum EmuError {
+    #[error("Only one instance of the QEMU Emulator is permitted")]
+    MultipleInstances,
+    #[error("QEMU emulator args cannot be empty")]
+    EmptyArgs,
+    #[error("Too many arguments passed to QEMU emulator ({0} > i32::MAX)")]
+    TooManyArgs(usize),
+}
+
 #[allow(clippy::unused_self)]
 impl Emulator {
     #[allow(clippy::must_use_candidate, clippy::similar_names)]
-    pub fn new(args: &[String], env: &[(String, String)]) -> Emulator {
+    pub fn new(args: &[String], env: &[(String, String)]) -> Result<Emulator, EmuError> {
         unsafe {
-            assert!(
-                !EMULATOR_IS_INITIALIZED,
-                "Only an instance of Emulator is permitted"
-            );
+            if EMULATOR_IS_INITIALIZED {
+                return Err(EmuError::MultipleInstances);
+            }
+        }
+        if args.is_empty() {
+            return Err(EmuError::EmptyArgs);
+        }
+
+        let argc = args.len();
+        if usize::try_from(i32::MAX).map_or(false, |max| argc > max) {
```

Comment:
It would be far easier to just do `i32::try_from(argc)` here, the try_from will already check against max.